pwnlib/heap/__init__.py

```diff
@@ -0,0 +1,356 @@
+# -*- coding: utf-8 -*-
+"""
+During heap exploit development, it is frequently useful to obtain an
+image of the heap layout as well as of the bins used by the glibc.
+
+
+To provide access to the heap items, an :class:`HeapExplorer` has to be used,
+which can be obtained from :attr:`pwnlib.tubes.process.heap_explorer`
+
+Examples
+----------
+
+Get a summary of the items of the arena:
+
+    >>> p = process('sh')
+    >>> hp = p.heap_explorer
+    >>> print(hp.arena().summary()) # doctest: +SKIP
+    ========================== Arena ==========================
+    - Malloc State (0x7fad3f0a5c40)
+        top = 0x564c90e90f20
+        last_remainder = 0x0
+        next = 0x7fad3f0a5c40
+        next_free = 0x0
+        system_mem = 0x21000
+    - Heap (0x564c90e83000)
+        chunks_count = 0x245
+        top: addr = 0x564c90e90f20, size = 0x130e0
+    - Tcaches
+        [23] 0x188 (1)
+        [41] 0x2a8 (1)
+    - Fast bins
+        [-] No chunks found
+    - Unsorted bins
+        [-] No chunks found
+    - Small bins
+        [-] No chunks found
+    - Large bins
+        [-] No chunks found
+    ===========================================================
+
+View the malloc state:
+
+    >>> p = process('sh')
+    >>> hp = p.heap_explorer
+    >>> print(hp.malloc_state()) # doctest: +SKIP
```

Comment:
Perhaps if we use `bash-static` and set `aslr=False` on the process, we can avoid skipping these doctests?

If these are not intended to be doctests, please use `.. code-block:: python` as mentioned above